src/purchase-client.ts

```diff
@@ -0,0 +1,203 @@
+import { MuxySDK } from ".";
+import { CurrentEnvironment } from './util';
+import Config from './config';
+
+import { TwitchBitsProduct, TwitchBitsTransaction } from './twitch';
+
+// Muxy dev variable
+declare global {
+  interface Window {
+    MEDKIT_PURCHASABLE_ITEMS: Product[] | PromiseLike<Product[]>;
+  }
+}
+
+export enum PurchaseClientType {
+  Dev,
+  Server,
+  Twitch,
+  Unknown
+}
+
+export interface PurchaseClient {
+  getProducts(): Promise<Product[]>;
+  purchase(identifier: string): void;
+  onUserPurchase(callback: (transaction: Transaction) => void): void;
+}
+
+export interface Cost {
+  amount: number;
+  type: string;
+}
+
+export interface Product {
+  sku: string;
+  displayName: string;
+  cost: Cost;
+}
+
+export interface Transaction {
+  transactionId: string;
+  product: Product;
+  userId: string;
+  displayName: string;
+  initiator: any;
+  transactionReceipt: string;
+}
+
+// TwitchPurchaseClient implements the basic 'purchase client' interface.
+// This is used by SDK to provide low-level access to twitch bits transactions.
+class TwitchPurchaseClient implements PurchaseClient {
+  /**
+   * purchase will start the twitch bits transaction.
+   *
+   * @since 2.4.0
+   *
+   * @throws {TwitchHelperError} Will throw an error if the Twitch Helper didn't load.
+   * 
+   * @param identifier the twitch sku of the item to be purchased.
+   *
+   * @example
+   * client.purchase("XXSKU000");
+   */
+  public purchase(identifier: string) {
+    if (!window.Twitch.ext.bits) {
```

Comment:
This check should have the optional chaining (`?.`) so as to not throw an error if one along the path is undefined.